pkg/client/dependency_graph.go

```diff
@@ -0,0 +1,62 @@
+package client
+
+import pkg "kcl-lang.io/kpm/pkg/package"
+
+// Construct dependency graph
+type DependencyGraph map[string][]string
+
+// Function to construct dependency graph by parsing kcl.mod file
+func ConstructDependencyGraphFromModFile(kpmClient *KpmClient, kclPkg *pkg.KclPkg) (DependencyGraph, error) {
+	dependencies, err := kpmClient.ParseKclModFile(kclPkg)
+	if err != nil {
+		return nil, err
+	}
+	return ConstructDependencyGraph(dependencies), nil
+}
+
+// Function to construct dependency graph from dependency map
+func ConstructDependencyGraph(dependencies map[string]map[string]string) DependencyGraph {
+	graph := make(DependencyGraph)
+	for dependency, details := range dependencies {
+		// Construct full module path including version or other attributes
+		fullPath := dependency
+		if version, ok := details["version"]; ok {
```

Comment:
Your unit test form is right. However, I don't think it covers all situations, such as Git dependencies.

Reply:
Ok.... So should I update it once again?

Reply:
See here https://kcl-lang.io/docs/next/user_docs/guides/package-management/how-to/kcl_mod#dependencies to change the code and update it in this PR.

Reply:
For the deeper deps, You can open another PR.

Reply:
You want me to change the unit test function?

Reply:
No, I think this line **does not** cover all situations. See https://kcl-lang.io/docs/next/user_docs/guides/package-management/how-to/kcl_mod#dependencies
```
if version, ok := details["version"]; ok {
```